src/pspc/iterator/FilmIteratorBase.h

```diff
@@ -0,0 +1,389 @@
+      std::string writeRGrid_;
+
+   };
+
+   // Inline member functions
+
+   // Return reference to iterator within this FilmIterator
+   template <int D, typename IteratorType>
+   inline IteratorType& FilmIteratorBase<D, IteratorType>::iterator()
+   { return iterator_; }
+
+   // Return const reference to iterator within this FilmIterator
+   template <int D, typename IteratorType>
+   inline IteratorType const & FilmIteratorBase<D, IteratorType>::iterator() const
+   { return iterator_; }
+
+   // Return whether unit cell is flexible.
+   template <int D, typename IteratorType>
+   inline const bool FilmIteratorBase<D, IteratorType>::isFlexible() 
+   { return iterator().isFlexible(); }
+
+   // Set value of chi between species s and wall w
+   template <int D, typename IteratorType>
+   inline void FilmIteratorBase<D, IteratorType>::setChi(int s, int w, double chi)
+   { chi_(s,w) =  chi; }
+
+   // Get const wall cField by reference
+   template <int D, typename IteratorType>
+   inline DArray<double> const & FilmIteratorBase<D, IteratorType>::wallCField() const
+   { return wallCField_; }
+
+   // Get value of normalVec
+   template <int D, typename IteratorType>
+   inline int FilmIteratorBase<D, IteratorType>::normalVec()
+   { return normalVec_; }
+
+   // Get const value of normalVec
+   template <int D, typename IteratorType>
+   inline int const FilmIteratorBase<D, IteratorType>::normalVec() const
+   { return normalVec_; }
+
+   // Get value of interfaceThickness
+   template <int D, typename IteratorType>
+   inline double FilmIteratorBase<D, IteratorType>::interfaceThickness()
+   { return t_; }
+
+   // Get const value of interfaceThickness
+   template <int D, typename IteratorType>
+   inline double const FilmIteratorBase<D, IteratorType>::interfaceThickness() const
+   { return t_; }
+
+   // Get value of wallThickness
+   template <int D, typename IteratorType>
+   inline double FilmIteratorBase<D, IteratorType>::wallThickness()
+   { return T_; }
+
+   // Get const value of wallThickness
+   template <int D, typename IteratorType>
+   inline double const FilmIteratorBase<D, IteratorType>::wallThickness() const
+   { return T_; }
+
+   // Get chi matrix by reference
+   template <int D, typename IteratorType>
+   inline DMatrix<double>& FilmIteratorBase<D, IteratorType>::chi()
+   { return chi_; }
+
+   // Get chi matrix by const reference
+   template <int D, typename IteratorType>
+   inline DMatrix<double> const & FilmIteratorBase<D, IteratorType>::chi() const
+   { return chi_; }
+
+   // Get the chi parameter between wall w and species s by reference
+   template <int D, typename IteratorType>
+   inline double& FilmIteratorBase<D, IteratorType>::chi(int s, int w)
+   { return chi_(s, w); }
+
+   // Get the const chi parameter between wall w and species s by reference
+   template <int D, typename IteratorType>
+   inline double const & FilmIteratorBase<D, IteratorType>::chi(int s, int w) const
```

Comment:
Oh, now I get it - you've allowed for different interactions for top and bottom walls !